src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -960,13 +961,24 @@ static inline bool notification_match(reservation_t& res,
 		      reservation_t& res,
 		      const RGWObjTags* req_tags)
 {
-  const RGWPubSub ps(res.store, res.user_tenant);
-  const RGWPubSub::Bucket ps_bucket(ps, res.bucket);
   rgw_pubsub_bucket_topics bucket_topics;
-  auto rc = ps_bucket.get_topics(res.dpp, bucket_topics, res.yield);
-  if (rc < 0) {
-    // failed to fetch bucket topics
-    return rc;
+  if (res.store->svc()->zone->notification_replication_supported()) {
```

Comment:
please separate the 2 features - new data model and multisite syncing

Reply:
> also, for the migration process, we would need this parameter to have 3 states:
> 
> * v1 - current data model, no replication
> * migrating - topic migration is in process, all topic/notification REST commands are blocked, topic replication should happen, as we will be creating the new topic objects in the migration process
> * v2 - new data model, replication is happening, lazy migration of notification configuration

i think during the migration we could live with `ceph.conf` flag indicating `migration` and the create/delete s3 request can look up at that flag and return 500